ReactNative/ReactTwitter/core/views/main-navigator.js

```diff
@@ -19,11 +23,76 @@ const OauthViewID = 'oauth-screen-identifier'
 const tweetsListID = 'tweets-list-identifier'
 const tweetViewID = 'tweet-view-identifier'
 
-var MainNavigator = React.createClass({
+var NavigationBarRouteMapper = {
+  LeftButton (route, navigator, index, navState) {
+    if (index === 0) {
+      return null
+    }
+
+    return (
+      <TouchableOpacity
+        onPress={() => navigator.pop()}
+        style={styles.navBarLeftButton}>
+        <Text style={[styles.navBarText, styles.navBarButtonText]}>
+          Back
+        </Text>
+      </TouchableOpacity>
+    )
+  },
+
+  RightButton (route, navigator, index, navState) {
+    if (index !== 0 || !__DEV__) { // eslint-disable-line no-undef
+      return null
+    }
+
+    return (<TouchableOpacity
+        onPress={() => navigator.push({
+          id: debugScreenID})}
+        style={styles.navBarRightButton}>
+            <Image style={styles.navBarIcon} source={require('./assets/debug.png')} />
+      </TouchableOpacity>
+    )
+  },
+
+  Title (route, navigator, index, navState) {
+    return (
+      <Text style={[styles.navBarText, styles.navBarTitleText]}>
+        {this._getTitleByScreenId(route.id)}
+      </Text>
+    )
+  },
+
+  _getTitleByScreenId (screenId) {
+    switch (screenId) {
+      case splashScreenID:
+        return 'SplashScreen'
```

Comment:
This is a really good point, as we are not considering localisation in this spike. I will add a task on jira for it